pyproject.toml

```diff
@@ -14,6 +14,11 @@ required-version = "23.3.0"
 # this feature is tracked in this issue:
 # https://github.com/psf/black/issues/2188
 preview = true
+exclude = '''
```

Comment:
Use `extend-exclude` so that other directories like `.git` remain excluded. Also, the patterns are checked recursively, so add `^/`. And should we just ignore all of `^/external`? This seems to work:

```
extend-exclude = '''
(
  ^/external
)
'''
```

Reply:
Okay, done! Thanks for the review :)